tests/matrix_free/fe_remote_evaluation_01.cc

```diff
@@ -0,0 +1,436 @@
+                phi_remote.reinit(cell->active_cell_index(), f);
+
+                for (unsigned int q : phi.quadrature_point_indices())
+                  {
+                    const auto             point = phi.quadrature_point(q);
+                    Tensor<1, dim, Number> temp;
+                    for (unsigned int i = 0; i < dim; ++i)
+                      temp[i] = point[i][v];
+
+                    error += temp - phi_remote.get_value(q);
+                  }
+              }
+          }
+        deallog << "    accumulated error: " << error << std::endl;
+      };
+
+      matrix_free.template loop<VectorType, VectorType>(
+        {}, {}, boundary_function, dst, src, true);
+    }
+}
+
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+  MPILogInitAll                    all;
+
+
+  deallog << "Setup MatrixFree..." << std::endl;
+  // 1) setup MatrixFree for different finite elements, i.e.
+  //    FESystem(FE_DGQ, dim + first_selected_component).
+
+  const unsigned int refinements = 2;
+  const unsigned int degree      = 5;
+
+  // Construct triangulation.
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+  GridGenerator::hyper_cube(tria);
+  tria.refine_global(refinements);
+
+  // Create DoFHandlers with different number of components to
+  // check different first_selected components
+  std::vector<std::unique_ptr<DoFHandler<dim>>> dof_handlers;
+  unsigned int                                  max_first_selected_comp = 3;
+  for (unsigned int i = 0; i < max_first_selected_comp; ++i)
+    {
+      dof_handlers.push_back(std::make_unique<DoFHandler<dim>>(tria));
+      dof_handlers.back()->distribute_dofs(
+        FESystem<dim>(FE_DGQ<dim>(degree), dim + i));
+    }
+
+  // Constraints are not considered in this test
+  AffineConstraints<Number> constraints;
+  constraints.close();
+
+  // Setup MatrixFree
+  typename MatrixFree<dim, Number>::AdditionalData data;
+  data.mapping_update_flags = update_values;
+  // We are not considering inner faces in this test. Therefore,
+  // data.mapping_update_flags_inner_faces is not set.
+  data.mapping_update_flags_boundary_faces =
+    update_quadrature_points | update_values;
+
+
+  // Construct vector of raw pointers needed to init matrix free
+  std::vector<const DoFHandler<dim> *>           dof_handlers_mf;
+  std::vector<const AffineConstraints<Number> *> constraints_mf;
+  for (auto &dh : dof_handlers)
+    {
+      dof_handlers_mf.push_back(dh.get());
+      constraints_mf.push_back(&constraints);
+    }
+
+  MatrixFree<dim, Number> matrix_free;
+  matrix_free.reinit(MappingQ1<dim>(),
+                     dof_handlers_mf,
+                     constraints_mf,
+                     QGauss<dim>(degree + 1),
+                     data);
+
+  // 2) test to access remote values at points in face batches
+  deallog << "\nTest the access of remote values at points in face batches..."
```

Comment:
Maybe "\n" is the problem?

Reply:
Out of curiosity, I replaced it by `std::endl`, but as one would expect, the same errors occurred.